### ISSUE TITLE
Changed error code with v1.2.0 

### ISSUE THREAD

**Issue by daeho-ro** [lima-vm/socket_vmnet#99]
Hello, we have a simple build test and it seems that the error code is changed from `1` to `139`.
Can you check this is intended and have to change our test?
Thanks,

```
  ==> /opt/homebrew/opt/socket_vmnet/bin/socket_vmnet /dev/null 2>&1
  sh: line 1:  7219 Segmentation fault: 11  /opt/homebrew/opt/socket_vmnet/bin/socket_vmnet /dev/null 2>&1
  WARNING: --vmnet-gateway=IP should be explicitly specified to avoid conflicting with other applications
  ERROR| WARNING: Running without root. This is very unlikely to work: See README.md
  ERROR| bind: Address already in use
  ERROR| socket_bindlisten: Address already in use
  Error: socket_vmnet: failed
  An exception occurred within a child process:
    Minitest::Assertion: Expected: 1
    Actual: 139
```

**Comment by jandubois** [comment]
This is not intentional, but I think you still want to modify your test because `/dev/null` is not a valid socket:

```diff
- /opt/homebrew/opt/socket_vmnet/bin/socket_vmnet /dev/null 2>&1
+ /opt/homebrew/opt/socket_vmnet/bin/socket_vmnet >/dev/null 2>&1
```

**Comment by jandubois** [comment]
Please leave this issue open though; the program should not crash, even with invalid input.

Old version:

```console
$ socket_vmnet /dev/null
WARNING: --vmnet-gateway=IP should be explicitly specified to avoid conflicting with other applications
WARNING: Running without root. This is very unlikely to work. See README.md .
bind: Address already in use
socket_bindlisten: Address already in use
```

New version:

```console
$ ./socket_vmnet /dev/null
WARNING: --vmnet-gateway=IP should be explicitly specified to avoid conflicting with other applications
ERROR| WARNING: Running without root. This is very unlikely to work: See README.md
ERROR| bind: Address already in use
ERROR| socket_bindlisten: Address already in use
Segmentation fault: 11
```

**Comment by jandubois** [comment]
There are other ways to get the program to crash:

```console
$ ./socket_vmnet -p /dev/null
Segmentation fault: 11
```

This crashes with both the old and the new version. It too should produce a proper error message instead.

**Comment by jandubois** [comment]
> ```
> WARNING: --vmnet-gateway=IP should be explicitly specified to avoid conflicting with other applications
> ERROR| WARNING: Running without root. This is very unlikely to work: See README.md
> ```

Also some feedback on the prefixes:

* Why do warnings have a "WARNING: " prefix but errors use "ERROR| "?
* And why does the first line have both prefixes: "ERROR| WARNING:"? What is it?

**Comment by nirs** [comment]
> > ```
> > WARNING: --vmnet-gateway=IP should be explicitly specified to avoid conflicting with other applications
> > ERROR| WARNING: Running without root. This is very unlikely to work: See README.md
> > ```
> 
> Also some feedback on the prefixes:
> 
> * Why do warnings have a "WARNING: " prefix but errors use "ERROR| "?
> * And why does the second line have both prefixes: "ERROR| WARNING:"? What is it?

The first warning come from cli.c, which does not use the logger.

The second one is an error, and the error test contains "Warning:". We need to add a WARNF() macro.

**Comment by nirs** [comment]
@jandubois is this fixed by #100?

**Comment by jandubois** [comment]
> @jandubois is this fixed by #100?

I believe the crashes are fixed with #100, so I'm going to close it. Will you create a new issue for the confusing warning/error messages?

**Comment by nirs** [comment]
> Will you create a new issue for the confusing warning/error messages?
Added #106 and #107
